runtime/lua/vim/lsp/health.lua

```diff
@@ -66,7 +66,7 @@ local function check_watcher()
   end
 
   report_info('File watch backend: ' .. watchfunc_name)
-  if watchfunc_name == 'libuv-watchdirs' then
+  if vim.lsp._watchfiles._is_os_supported() and watchfunc_name == 'libuv-watchdirs' then
```

Comment:
watchdirs isn't used on windows or Mac, see https://github.com/neovim/neovim/blob/080cd39d39539d53d5450ab585e5d9324c660157/runtime/lua/vim/lsp/_watchfiles.lua#L10

And users can still opt in to file watching. If we want to hide it if it's disabled we'd have to check active clients and see if it's disabled on all

Reply:
Lmk if this is what you meant.

Reply:
Wait I think I got that wrong since `capabilities` refers to Neovim's capabilities and not the server's. However I cannot find the respective server capability for `didChangeWatchedFiles`, so how exactly can we check if it's enabled?

Reply:
Using the client capabilities is fine here. Users can change them.

E.g. I have:

```lua
  local capabilities = vim.tbl_deep_extend(
    "force",
    lsp.protocol.make_client_capabilities(),
    lsp_compl.capabilities(),
    {
      workspace = {
        didChangeWatchedFiles = {
          dynamicRegistration = true
        }
      },
    }
  )
```

and provide them to `vim.lsp.start`.
If we also wanted to check if a server makes use of the capability, we'd have to check `client.dynamic_capabilities:get("workspace/didChangeWatchedFiles")`